syntax = "proto3";

import "gymize.proto";

option csharp_namespace = "Gymize.Protobuf";

enum MessageTypeProto {
	MESSAGE_TYPE_PROTO_UNSPECIFIED = 0;
	MESSAGE_TYPE_PROTO_MESSAGE = 1;
	MESSAGE_TYPE_PROTO_REQUEST = 2;
	MESSAGE_TYPE_PROTO_RESPONSE = 3;
}

message HeaderProto {
	MessageTypeProto message_type = 1;
	string id = 2;
	bytes uuid = 3;
}

message ContentProto {
	oneof data {
		bytes raw = 1;
		string text = 2;
		GymizeProto gymize = 3;
	}
}

message MessageProto {
	HeaderProto header = 1;
	ContentProto content = 2;
}